pkg/apis/core/validation/shoot.go

```diff
@@ -405,6 +405,20 @@ func ValidateNodeCIDRMaskWithMaxPod(maxPod int32, nodeCIDRMaskSize int32) field.
 	return allErrs
 }
 
+// validateTotalNodeCountWithPodCIDR validates if the Pod Network has enough ip addresses (configured via PodCIDR on the network config and the NodeCIDRMask on the kube controller manager) to support the total number of nodes from the worker pools of the shoot
+func validateTotalNodeCountWithPodCIDR(totalNodes int32, nodeCIDRMaskSize int32, podNetworkCIDR string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	CIDR, _ := strconv.Atoi(podNetworkCIDR[len(podNetworkCIDR)-2:])
+	// first and last IPs are reserved
+	maxNodeCount := int32(math.Pow(2, float64(nodeCIDRMaskSize-int32(CIDR))) - 2)
+
+	if totalNodes > maxNodeCount {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("provider").Child("workers"), totalNodes, fmt.Sprintf("worker configuration incorrect. The spec.networking.pod only supports %d IP addresses. The total of all the nodes specified by the worker pool should be less than %d ", maxNodeCount, maxNodeCount)))
```

Comment:
`maxNodeCount` is not the number of IP addresses. I do not understand the error message. Please adapt the second sentence so that it makes sense.